app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,28 @@
+class MessagesController < ApplicationController
+  before_action :load_permissions
+  load_and_authorize_resource :group, parent: true
+  load_and_authorize_resource :message
+
+  def index
+    @messages = @group.messages.by_latest.includes(user: :first_post, message_comments: :user)
```

Comment:
My GIT was really messing with me for this PR, redid some changes three times...